webapp/src/js/components/Plot/Table/Actions.js

```diff
@@ -87,7 +90,14 @@ let TablePlotActions = React.createClass({
       label: table.capNamePlural
     }));
 
-
+    this.definedQuery = query;
```

Comment:
In some cases, e.g. DataTableWithActions, I think I was trying to avoid working out the same value more than once, or passing it around as an argument, within the same component.

Yeah, the whole approach is somewhat unsatisfactory... Any ideas?